File: src/main/java/ac/dnd/server/common/support/FieldError.java
package ac.dnd.server.common.support;

import java.util.List;

import org.springframework.validation.BindingResult;

public record FieldError(
	String field,
	String value,
	String reason
) {
	public static List<FieldError> of(
		final String field,
		final String value,
		final String reason
	) {
		return List.of(new FieldError(
			field,
			value,
			reason
		));
	}

	public static List<FieldError> of(
		final BindingResult bindingResult
	) {
		final List<org.springframework.validation.FieldError> fieldErrors = bindingResult.getFieldErrors();
		return fieldErrors.stream()
			.map(error -> new FieldError(
				error.getField(),
				error.getRejectedValue() == null ? "" : error.getRejectedValue().toString(),
				error.getDefaultMessage()
			))
			.toList();
	}
}

File: src/main/kotlin/ac/dnd/server/common/support/FieldError.kt
package ac.dnd.server.common.support

import org.springframework.validation.BindingResult

data class FieldError(
    val field: String,
    val value: String,
    val reason: String
) {
    companion object {
        fun of(field: String, value: String, reason: String): List<FieldError> {
            return listOf(FieldError(field, value, reason))
        }

        fun of(bindingResult: BindingResult): List<FieldError> {
            val fieldErrors = bindingResult.fieldErrors
            return fieldErrors.map { error ->
                FieldError(
                    error.field,
                    error.rejectedValue?.toString() ?: "",
                    error.defaultMessage ?: ""
                )
            }
        }
    }
}